requests/resolver.go

```diff
@@ -34,10 +34,28 @@ func NewResolverMap(hostToIP map[string][]string) ResolveContext {
 		}
 
 		addrs := []string{addr}
+
+		for _, addrIP := range FetchDNSFromCache(addrHost) {
+			addrs = append(addrs, net.JoinHostPort(addrIP, addrPort))
+		}
+
 		for _, addrIP := range hostToIP[addrHost] {
 			addrs = append(addrs, net.JoinHostPort(addrIP, addrPort))
 		}
 
-		return addrs, nil
+		return deduplicate(addrs), nil
+	}
+}
+
+func deduplicate(list []string) (result []string) {
```

Comment:
This could probably go into a common utils package. Do we have such thing?

Reply:
Yeah, we have such a package, but there is no point to move it to a common package while it's needed only in one place. We moving it after it starts to be needed in several places.